### ISSUE TITLE
Playground server 500 on PHP 8.1/WP6.3

### ISSUE THREAD

**Issue by ipajen** [aspirepress/AspireUpdate#58]
1. Go to https://playground.wordpress.net/?blueprint-url=https://raw.githubusercontent.com/aspirepress/AspireUpdate/refs/heads/playground-ready/assets/playground/blueprint.json

2. In the playground settings choose WP 6.3 and PHP 8.1 and click apply settings

3. playground  gives   "httpStatusCode": 500,

Expected that it should work to change to WP 6.3 and PHP 8.1

**Comment by ipajen** [comment]
Found this In the playground logs

`27-Oct-2024 23:12:21 UTC] JavaScript Error: Theme twentytwentyfour could not be activated – WordPress exited with no error. Sometimes, when $_SERVER or site options are not configured correctly, WordPress exits early with a 301 redirect. Inspect the "debug" logs in the console for more details
@https://playground.wordpress.net/assets/config-380f8aaf.js:55:111
[27-Oct-2024 23:12:21 UTC] JavaScript Error: Error when executing the blueprint step #7 ({"step":"installTheme","themeData":{"resource":"wordpress.org/themes","slug":"twentytwentyfour"}}) : Theme twentytwentyfour could not be activated – WordPress exited with no error. Sometimes, when $_SERVER or site options are not configured correctly, WordPress exits early with a 301 redirect. Inspect the "debug" logs in the console for more details
@https://playground.wordpress.net/assets/config-380f8aaf.js:750:145943`

Twenty Twenty-Four Came in WP 6.4,  would it better to use 2023 theme instead as default ?

**Comment by ipajen** [comment]
Twenty Twenty-Two WP 5.9
Twenty Twenty-Three WP 6.1

**Comment by namithj** [comment]
This looks like a wordpress bug and not related to our plugin.

**Comment by ipajen** [comment]
> This looks like a wordpress bug and not related to our plugin.

Can you explain how its a WP bug? As how I see it its that we use the twentytwentyfour in the blueprint that I guess its not compatible with older WP versions. For testing purpose it would be better to use the Twenty Twenty-Two so its possible to test with WP 5.9 and above and AP.

**Comment by namithj** [comment]
Its something related to 2024 and 2022 right, unrelated to our plugin.

**Comment by ipajen** [comment]
It’s we who define it in the blueprint 

https://github.com/aspirepress/AspireUpdate/blob/9f14298cc8fa7778b9d6b157db8e7ad1c8b7f57d/assets/playground/blueprint.json#L48

1.	Twenty Twenty-Two: This theme was designed to work with WordPress 5.9 and above, given its full integration with the block editor features introduced in that version.

2.	Twenty Twenty-Three: This theme requires at least WordPress 6.1, aligning with updates in the block editor and design tools.

3.	Twenty Twenty-Four: The newest default theme, it is built to work with WordPress 6.4, taking advantage of all the latest tools and patterns in WordPress’s site editing features

**Comment by namithj** [comment]
@asirota should be able to update blueprint.json to reflect that, assigning to @asirota

**Comment by asirota** [comment]
Yup it may be a WordPress playgrounds issue. Need to investigate. I doubt it's a bug in our plugin.

**Comment by asirota** [comment]
Fixed @ipajen - try AspireUpdate in WordPress playground with 2021 theme. 6.3 in the earliest WordPress supported. Maybe they'll make older WordPress available eventually to try. That would be nice.

**Comment by ipajen** [comment]
@asirota Ir would be better to use 2022 as https://wordpress.github.io/wordpress-playground/blueprints/examples/#load-an-older-wordpress-version 

Note: the oldest supported WordPress version is 5.9.9, following the SQLite integration plugin.

if using to old version of theme it could be problematic when newer PHP or WP versions comes.

**Comment by asirota** [comment]
Our plugin supports 5.4 as per the readme.

**Comment by ipajen** [comment]
> Our plugin supports 5.4 as per the readme.

The minimum version today is set to WP 5.3 of AspireUpdate? Is it worth supporting this old version ?  https://wordpress.org/about/stats/ as Playground minimum version is **WP 5.9.9**, Query Monitor WP 5.8 and GitUpdater WP 5.9. As of testing point of view becomes more difficult without my tools to use :)   As if using 2021 you can’t still test WP 5.4-5.8 in playground.

**Comment by asirota** [comment]
That's ok. Our user base is the concern. WordPress playgrounds set 5.9.9 to be in sync with WordPress but you can manually invoke an older version just fine using the link approach or modifying the blueprint.

**Comment by ipajen** [comment]
Note: the oldest supported WordPress version is 5.9.9, following the SQLite integration plugin.

https://wordpress.github.io/wordpress-playground/blueprints/examples/#load-an-older-wordpress-version

so if I understand you correctly it’s possible to use WP 5.3 on playground ?

**Comment by asirota** [comment]
Yes. Using the query parameter or a blueprint.JSON change. Docs available
on WordPress playground website.

Alex Sirota, PMP                 PGP Public Key <https://bit.ly/asirotapgp>
Director at NewPath Consulting <https://newpathconsulting.com/>

Start a free WildApricot trial.
<https://register.wildapricot.com/?refcode=M6QF7KFQ6M>
Try Make.com for free. <https://www.make.com/en/register?pc=newpath1>

**Comment by asirota** [comment]
Looks like playground indeed only works with 5.9.9 of wordpress or later, I will change the blueprint.json to use 2022 theme (twentytwentytwo). Committed to playground-ready.
